src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -2094,5 +2095,17 @@ internal void ReleaseAcceptingNewBlocks()
                 }
             }
         }
+        
+        public void ForkChoiceUpdated(Keccak? HeadBlockHash, Keccak? finalizedBlockHash, Keccak? safeBlockHash)
+        {
+            FinalizedHash = finalizedBlockHash;
+            SafeHash = safeBlockHash;
+            using (_metadataDb.StartBatch())
+            {
+                _metadataDb.Set(MetadataDbKeys.FinalizedBlockHash, FinalizedHash!.Bytes);  
+                _metadataDb.Set(MetadataDbKeys.SafeBlockHash, SafeHash!.Bytes);  
+            }
+            SetHeadBlock(HeadBlockHash);
```

Comment:
should `FinalizedHash` and `SafeHash` be updated the same way as the `SetHeadBlock` function?